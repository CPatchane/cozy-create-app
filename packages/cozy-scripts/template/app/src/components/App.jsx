import React from 'react'
import { Route, Switch, Redirect, HashRouter } from 'react-router-dom'

import Sidebar from './Sidebar'
import Todos from './Todos'
import Hello1 from './HelloViews/Hello1'
import Hello2 from './HelloViews/Hello2'

const App = () => (
  <HashRouter>
    <div className="app-wrapper o-layout--2panes">
      <Sidebar />
      <main className="app-content">
        <Switch>
          <Route path="/todos" component={Todos} />
          <Route path="/viewhello1" component={Hello1} />
          <Route path="/viewhello2" component={Hello2} />
          <Redirect from="/" to="/todos" />
          <Redirect from="*" to="/todos" />
        </Switch>
      </main>
    </div>
  </HashRouter>
)

export default App
