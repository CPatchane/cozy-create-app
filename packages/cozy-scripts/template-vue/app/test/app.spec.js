'use strict'

/* eslint-env jest */

import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'

import App from '../src/components/App'
import I18nTestMixin from './lib/I18n'
Vue.mixin(I18nTestMixin)

describe('App component only', () => {
  it('should be mounted correctly', () => {
    const component = shallowMount(App)
    expect(component.html()).toMatchSnapshot()
  })
})
