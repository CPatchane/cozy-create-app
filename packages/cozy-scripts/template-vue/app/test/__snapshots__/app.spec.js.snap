// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component only should be mounted correctly 1`] = `
<div class="app-wrapper o-layout-2panes">
  <aside class="o-sidebar">
    <nav>
      <ul class="c-nav">
        <li class="c-nav-item">
          <router-link-stub to="/viewhello1" tag="a" event="click" class="c-nav-link">
            <icon-stub icon="test-file-stub" iconwidth="1em" color="" iconheight="1em" iconclass="" class="c-nav-icon"></icon-stub>
            First
          </router-link-stub>
        </li>
        <li class="c-nav-item">
          <router-link-stub to="/viewhello2" tag="a" event="click" class="c-nav-link">
            <icon-stub icon="test-file-stub" iconwidth="1em" color="" iconheight="1em" iconclass="" class="c-nav-icon"></icon-stub>
            Second
          </router-link-stub>
        </li>
        <li class="c-nav-item">
          <router-link-stub to="/viewhello3" tag="a" event="click" class="c-nav-link">
            <icon-stub icon="test-file-stub" iconwidth="1em" color="" iconheight="1em" iconclass="" class="c-nav-icon"></icon-stub>
            Third
          </router-link-stub>
        </li>
      </ul>
    </nav>
  </aside>
  <main>
    <div class="app-content">
      <router-view-stub name="default" class="view"></router-view-stub>
    </div>
  </main>
</div>
`;
